platform/posix/transport/src/sockets_posix.c

```diff
@@ -181,48 +260,12 @@ static SocketStatus_t attemptConnection( struct addrinfo * pListHead,
             continue;
         }
 
-        pAddrInfo = pIndex->ai_addr;
-
-        if( pAddrInfo->sa_family == AF_INET )
-        {
-            /* Store IPv4 in string to log. */
-            ( ( struct sockaddr_in * ) pAddrInfo )->sin_port = netPort;
-            addrInfoLength = sizeof( struct sockaddr_in );
-            ( void ) inet_ntop( pAddrInfo->sa_family,
-                                &( ( struct sockaddr_in * ) pAddrInfo )->sin_addr,
-                                resolvedIpAddr,
-                                sizeof( resolvedIpAddr ) );
-        }
-        else
-        {
-            /* Store IPv6 in string to log. */
-            ( ( struct sockaddr_in6 * ) pAddrInfo )->sin6_port = netPort;
-            addrInfoLength = sizeof( struct sockaddr_in6 );
-            ( void ) inet_ntop( pAddrInfo->sa_family,
-                                &( ( struct sockaddr_in6 * ) pAddrInfo )->sin6_addr,
-                                resolvedIpAddr,
-                                sizeof( resolvedIpAddr ) );
-        }
-
-        LogDebug( ( "Attempting to connect to server: Host=%.*s, IP address=%s.",
-                    ( int32_t ) hostNameLength,
-                    pHostName,
-                    resolvedIpAddr ) );
-
-        connectStatus = connect( *pTcpSocket, pAddrInfo, addrInfoLength );
+        /* Attempt to connect to a resolved DNS address of the host. */
+        returnStatus = connectToAddress( pIndex->ai_addr, pHostName, hostNameLength, port, *pTcpSocket );
 
-        if( connectStatus == -1 )
+        /* If connected to an IP address successfully, exit from the loop. */
+        if( returnStatus == SOCKETS_SUCCESS )
         {
-            LogWarn( ( "Failed to connect to server: Host=%.*s, IP address=%s.",
-                       ( int32_t ) hostNameLength,
-                       pHostName,
-                       resolvedIpAddr ) );
-            close( *pTcpSocket );
-        }
-        else
-        {
-            LogDebug( ( "Connected to IP address: %s.",
-                        resolvedIpAddr ) );
             break;
         }
 
```

Comment:
(Unrelated to the PR) Shouldn't these retry attempts use exponential back-off (with the `transport_reconnect` utility) ?

Reply:
Nevermind, I see the attempts are across different DNS addresses here instead of re-attempts to the same address